### ISSUE TITLE
Issues with RuleTester using eslint v9 flat config

### ISSUE THREAD

**Issue by awaismalik47** [angular-eslint/angular-eslint#2056]
Still getting the same issue

![Screenshot 2024-09-24 at 1 21 33 PM](https://github.com/user-attachments/assets/1c68e3b6-a92a-46dd-b0a8-7a5412069499)
![Screenshot 2024-09-24 at 10 10 33 AM](https://github.com/user-attachments/assets/52c09e1a-9d54-4cb8-8fb9-f31eab1cedeb)
![Screenshot 2024-09-24 at 10 11 12 AM](https://github.com/user-attachments/assets/8c2b1d2e-c2f7-4d45-adea-1cc32551763a)

**Comment by JamesHenry** [comment]
You should be using RuleTester from `@typescript-eslint/rule-tester`, you are currently importing it from utils

**Comment by awaismalik47** [comment]
okay sure let me share my the Error there

**Comment by JamesHenry** [comment]
@awaismalik47 Did my suggestion fix the issue?

**Comment by awaismalik47** [comment]
Not at all?

for the time we fixed it by adding this code 

import { RuleTester } from '@typescript-eslint/rule-tester';
const ruleTester = new RuleTester();

this actually cause error like structure clone 

so for this we add this line of code 

if ( typeof structuredClone === 'undefined' ) {
	global.structuredClone = (value) => {
		return require('v8').deserialize(require('v8').serialize(value));
	};
}


and yes it is working good 
but still we are in search to solve this issue with adding the structure clone code

**Comment by JamesHenry** [comment]
Create a js file, e.g. `debug.js` with the following contents:

```js
console.log(process.versions);
// Example output:
// 
// {
//   node: '20.17.0',
//   acorn: '8.11.3',
//   ada: '2.9.0',
//   ares: '1.32.3',
//   base64: '0.5.2',
//   brotli: '1.1.0',
//   cjs_module_lexer: '1.2.2',
//   cldr: '45.0',
//   icu: '75.1',
//   llhttp: '8.1.2',
//   modules: '115',
//   napi: '9',
//   nghttp2: '1.61.0',
//   nghttp3: '0.7.0',
//   ngtcp2: '1.1.0',
//   openssl: '3.0.13+quic',
//   simdutf: '5.3.0',
//   tz: '2024a',
//   undici: '6.19.2',
//   unicode: '15.1',
//   uv: '1.46.0',
//   uvwasi: '0.0.21',
//   v8: '11.3.244.8-node.23',
//   zlib: '1.3.0.1-motley-209717d'
// }

// Does not throw with the above output showing node 20
console.log(structuredClone);
```

And run it with node:

```sh
node debug.js
```

What results do you get?

**Comment by awaismalik47** [comment]
I got this output

![Screenshot 2024-10-02 at 10 54 58 AM](https://github.com/user-attachments/assets/4db79c2c-d1ea-4a29-9163-53ac2e9a1acf)

**Comment by JamesHenry** [comment]
Right so structuredClone does exist in your environment... Are you definitely seeing the rule tester error in the _same_ environment? Or are you only seeing it on CI for example?

**Comment by awaismalik47** [comment]
Currently i am facing this error in my local environment

**Comment by JamesHenry** [comment]
Ok, then hopefully it should be straightforward to reproduce. Please work on a minimal reproduction repo that fails with the error on your machine and push that up for me/others to take a look at and compare

**Comment by JamesHenry** [comment]
A lead on this - are you using jest with JSDOM? structuredClone is not defined in that environment.

https://github.com/jsdom/jsdom/issues/3363#issuecomment-1467894943

**Comment by JamesHenry** [comment]
I'm going to close this one, as I think it's fairly conclusive that this is a non-standard environment, most likely provided by a test runner, most likely configured to use JSDOM. Therefore, there are no changes applicable to angular-eslint.

Please feel free to reopen if you can reproduce in plain nodejs.

**Comment by True-Evzheniy** [comment]
`  testEnvironment: 'node',` should be added to default export from jest.config.ts to avoid structuredClone issue
